File: 2021/07/30/main.py
class MapSum:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.mp = {}
        

    def insert(self, key: str, val: int) -> None:
        self.mp[key] = val

    def sum(self, prefix: str) -> int:
        s, l = 0, len(prefix)
        for k, v  in self.mp.items():
            if k[:l] == prefix:
                s += v
        return s


# Your MapSum object will be instantiated and called as such:
# obj = MapSum()
# obj.insert(key,val)
# param_2 = obj.sum(prefix)

File: 2021/07/30/main.go
package main

type MapSum struct {
	mp map[string]int
}

/** Initialize your data structure here. */
func Constructor() MapSum {
	return MapSum{mp: make(map[string]int)}
}

func (this *MapSum) Insert(key string, val int) {
	this.mp[key] = val
}

func (this *MapSum) Sum(prefix string) int {
	s, l := 0, len(prefix)
	for k, v := range this.mp {
		if len(k) >= len(prefix) && k[:l] == prefix {
			s += v
		}
	}
	return s
}

/**
 * Your MapSum object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Insert(key,val);
 * param_2 := obj.Sum(prefix);
 */
